src/framework/components/sound/component.js

```diff
@@ -0,0 +1,372 @@
+                var slot = slots[key];
+                // play if autoPlay is true or
+                // if the slot was paused when the component
+                // got disabled
+                if (slot.autoPlay && slot.isStopped) {
+                    slot.play();
+                } else if (playingBeforeDisable[key]) {
+                    slot.resume();
+                } else if (!slot.isLoaded) {
+                    // start loading slots
+                    slot.load();
+                }
+            }
+        },
+
+        onDisable: function () {
+            SoundComponent._super.onDisable.call(this);
+
+            var slots = this.data.slots;
+            var playingBeforeDisable = {};
+            for (var key in slots) {
+                // pause non-overlapping sounds
+                if (! slots[key].overlap) {
+                    if (slots[key].isPlaying) {
+                        slots[key].pause();
+                        // remember sounds playing when we disable
+                        // so we can resume them on enable
+                        playingBeforeDisable[key] = true;
+                    }
+                }
+            }
+
+            this.data.playingBeforeDisable = playingBeforeDisable;
+        },
+
+        /**
+         * @function
+         * @param {[type]} name    [description]
+         * @param {[type]} options [description]
+         */
+
+        /**
+         * @function
+         * @name pc.SoundComponent#addTime
+         * @description Create a {@link pc.SoundSlot}
+         * @param {String} name The name of the slot
+         * @param {Object} options Settings for the slot
+         * @param {Number} [options.volume=1] The playback volume, between 0 and 1.
+         * @param {Number} [options.pitch=1] The relative pitch, default of 1, plays at normal pitch.
+         * @param {Boolean} [options.loop=false] If true the sound will restart when it reaches the end.
+         * @param {Number} [options.startTime=0] The start time from which the sound will start playing.
+         * @param {Number} [options.duration=null] The duration of the sound that the slot will play starting from startTime.
+         * @param {Boolean} [options.overlap=false] If true then sounds played from slot will be played independently of each other. Otherwise the slot will first stop the current sound before starting the new one.
+         * @param {Boolean} [options.autoPlay=false] If true the slot will start playing as soon as its audio asset is loaded.
+         * @param {Number} [options.asset=null] The asset id of the audio asset that is going to be played by this slot.
+         * @returns {pc.SoundSlot} The new slot.
+         */
+        addSlot: function (name, options) {
+            var slots = this.data.slots;
+            if (slots[name]) {
+                logERROR('A sound slot with name ' + name + ' already exists on Entity ' + this.entity.getPath());
+                return slots[name];
```

Comment:
We probably want to return `null` here to allow users to get result of their function, instead of double return value meaning (new slot, or existing one - this just complicates).
And do a `logWARNING`, as this is not exception case - this scenario is totally handled.